plugin/daemon_example/daemon_example.cc

```diff
@@ -181,7 +183,7 @@ static int daemon_example_plugin_deinit(void *p)
   struct tm tm_tmp;
   void *dummy_retval;
 
-  my_thread_cancel(&con->heartbeat_thread);
+  my_atomic_store32(&con->done, 1);
 
```

Comment:
Shouldn't we add `my_thread_join(&con->heartbeat_thread, nullptr);` after setting `done` to `1`?

Reply:
The `join` is actually already there a few lines below